MODULE.bazel

```diff
@@ -40,6 +40,54 @@ bazel_dep(name = "xxhash", version = "0.8.2")
 bazel_dep(name = "yaml-cpp", version = "0.8.0")
 bazel_dep(name = "zlib", version = "1.3")
 bazel_dep(name = "zstd", version = "1.5.6")
+bazel_dep(name = "toolchains_llvm", version = "1.0.0")
+
+# ====================================
+# LLVM toolchains
+# ====================================
+LLVM_VERSIONS = {"": "16.0.4"}
+BASE_COMPILER_FLAGS = {"": {
+}}
+
+# toolchains_llvm module
+# TODO: for now uses my private fork that allow us to disable linking against libunwind
+git_override(
+    module_name = "toolchains_llvm",
+    commit = "24b42c4db17f89bb4e829c0629ff5ea3b5291bce",
+    remote = "https://github.com/mmaslankaprv/toolchains_llvm",
+)
+
+llvm = use_extension("@toolchains_llvm//toolchain/extensions:llvm.bzl", "llvm")
+
+# base LLVM toolchain, this one will download the LLVM binaries specific for the host platform
+llvm.toolchain(
+    name = "llvm_toolchain",
+    llvm_versions = LLVM_VERSIONS,
+    cxx_standard = {"": "c++20"},
+    stdlib = {"": "builtin-libc++"},
+    use_libunwind = {"": "false"},
+)
+
+use_repo(llvm, "llvm_toolchain", "llvm_toolchain_llvm")
+
+register_toolchains("@llvm_toolchain//:all")
+
+# toolchain using system LLVM
+# TODO: submit a llvm_toolchain PR to exlcude libunwind
+llvm.toolchain(
```

Comment:
What is the motivation for this? We could just let the automatic bazel configuration take care of this and you can enable disable the bazel provided toolchain via --config flags (which everyone should opt into with a .user.bazelrc)

Reply:
We wanted to have an option to point bazel to use the system installed LLVM instead of downloading one. I've used the guidelines coming from:  https://github.com/bazel-contrib/toolchains_llvm?tab=readme-ov-file#bring-your-own-llvm.

Reply:
That's only needed if it's not installed on the system.

I think the recommended way would be to make the system provided llvm optional and have a config that people can enable by default to enable, and we can recommend everyone do that (or maybe there is a flag to enable the default toolchain to have priority with a flag, let me look). Practically that looks like removing the toolchain registration here and instead passing in the extra toolchains flag as mentioned here: https://bazel.build/versions/7.2.0/extending/toolchains#registering-building-toolchains

Reply:
I also think you could pass in `--extra_toolchains=@bazel_tools//tools/cpp:toolchain` and I think the default host tool chain would override the registered on in MODULE.bazel that can be configured with the options that Noah had and were deleted, and you can add that behind some --config=system_compiler